docs/overview/architecture.md

```diff
@@ -0,0 +1,240 @@
+2. Vald Ingress will forward the request to Vald Filter Gateway.
+3. Vald Filter Gateway will forward the request to Vald Ingress Filter.
+4. Vald Ingress Filter will perform the filtering and return the filtering result to the Vald Filter Gateway.
+5. Vald Filter Gateway will forward the request to the Vald Meta Gateway.
+6. Vald Meta Gateway will forward the request to the Vald Backup Gateway.
+7. Vald Backup Gateway will forward the request to the Vald LB Gateway.
+8. Vald LB Gateway will decide which Vald Agent instance to process the request base on the node resource usage and forwards the request to the decided Vald Agent.
+9. Vald Agent returns the searching result to the Vald LB Gateway. The searching result includes the UUID.
+10. Vald LB Gateway returns the searching result to Vald Backup Gateway.
+11. Vald Backup Gateway returns the searching result to the Vald Meta Gateway.
+12. Vald Meta Gateway will forward the searching result to the Vald Meta.
+13. Vald Meta will perform a search for the Vector ID base on the UUID and return the Vector ID to the Vald Meta Gateway.
+14. Vald Meta Gateway returns the searching result with the vector ID to the Vald Filter Gateway.
+15. Vald Filter Gateway will forward the request to Vald Egress Gateway to filter the final result.
+
+<!-- ### Update -->
+
+<!-- ### Delete -->
+
+## Components
+
+### Vald Filter
+
+Vald Filter is an optional functionality in Vald.
+User can implement the custom filtering logic and integrate with Vald.
+
+Vald Filter provides the following functionalities.
+
+- Custom filter based on request query
+- Custom filter for the searching result
+
+#### Vald Ingress Filter
+
+Vald Ingress Filter filters the incoming request before processing it.
+
+Users can implement custom filtering logic such as changing the vectors or filtering based on user ID.
+
+#### Vald Egress Filter
+
+Vald Egress Filter filters the response before sending it to the user.
+
+This component can reorder the searching result from multiple Vald Agents based on the user-defined ranking.
+
+#### Vald Filter Gateway
+
+Vald Filter Gateway forwards the request to Vald Ingress Filter before processing it and forwards the response to the Vald Egress Filter before returning the searching result to the user.
+
+### Vald Metadata
+
+In Vald, metadata consists of the vector data and the corresponding additional data to represent the set of the searching criteria and the result.
+
+Vald Metadata includes the user inputted metadata(vector ID) and the vector, and the internal generated UUID.
+
+#### Vald Meta Gateway
+
+The main responsibility of the Vald Meta Gateway is to process the Vald metadata and to forward the information to Vald Backup Gateway.
+
+It will perform the following action:
+
+1. Return error if the user has already input the same vector in Vald
+1. Generate the corresponding UUID for internal use.
+1. Forward the metadata (vector ID and UUID) request to the Vald Meta Agent.
```

Comment:
I think it is more clear, what do you think?
```suggestion
1. Forward the vector ID and UUID request to the Vald Meta.
```

Reply:
I think it is good. :+1: